### PR TITLE
stubgen: fixed handling of Protocol and added testcase

### LINKED ISSUE #12072: `stubgen` removes `Protocol` from base classes list
**Bug Report**

Consider the following input:
```python
from typing import Protocol

class Interface(Protocol):
    def dummy(self, arg: str) -> str:
        ...
```
`stubgen` produces:
```python
class Interface:
    def dummy(self, arg: str) -> str: ...
```

Note that the `Protocol` class is removed from the list of base classes.

Obviously this causes problems when performing type checking against users of that generated typestub:

```python
from interface import Interface

class InterfaceImpl:
    def dummy(self, arg: str) -> str:
        return arg


interf: Interface = InterfaceImpl()
```
mypy:
```
provider/__init__.py:8: error: Incompatible types in assignment (expression has type "InterfaceImpl", variable has type "Interface")
```

**Expected Behavior**

`Protocol` base class is not removed.

**Actual Behavior**

`Protocol` base class is removed.

**Your Environment**

- Mypy version used: mypy 0.931
- Python version used: Python 3.9.7
- Operating system and version: macOS

### PR DESCRIPTION
**Pull request by citruz**
### Description

This PR fixes #12072 by correctly handling `Protocol` definitions. Previously, the `Protocol` base class was removed when generating type stubs which causes problems with other packages that want to use type definition (because they see it as a regular class, not as a `Protocol`).

## Test Plan

Added a testcase to the stubgen testset.

### CODE REVIEW

File: test-data/unit/stubgen.test [hunk 1]
```diff
@@ -2580,3 +2580,18 @@ class A:
 def f(x: int, y: int) -> int: ...
 @t.overload
 def f(x: t.Tuple[int, int]) -> int: ...
+
+
+[case testProtocol_semanal]
+from typing import Protocol
+
+class P(Protocol):
```
Comment (sobolevn) [inline test-data/unit/stubgen.test:2588]: Let's also add a generic `Protocol[T]` class here 🤔

**Review by sobolevn — COMMENTED** [review]
Thank you!

File: test-data/unit/stubgen.test (hunk 1 above)
Comment (citruz) [reply test-data/unit/stubgen.test:2588]: You mean like this?
```python
T = TypeVar('T')
class PT(Protocol[T]):
    def f(self, x: T) -> T:
        ...
```
That results in:
```python
  T = TypeVar('T')

  class PT(Protocol):
      def f(self, x: T) -> T: ...
```
So I guess that needs to be handled explicitly in my implementation?

Comment (sobolevn) [reply test-data/unit/stubgen.test:2588]: Yes 🙂

Comment (citruz) [reply test-data/unit/stubgen.test:2588]: Done ✅

**Review by sobolevn — APPROVED** [review]
Thank you! 👍

**Comment by sobolevn** [comment]
@srittau do you want to double check this? I've seen multiple your PRs to `stubgen` 🙂

**Comment by citruz** [comment]
Any updates on this? :)

**Review by raghavthind2005 — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
If you fix the merge conflict I can review it and hopefully merge it.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**